### DISCUSSION TITLE
Window size & dynamical rendering

### DISCUSSION THREAD

**Discussion by ComradeMashkov** [vsg-dev/VulkanSceneGraph#1224] [Q&A]
Hey, vsg community! I study Vulkan and VulkanSceneGraph and now I have some confusing questions concerning window size.

1. How can I set window minimal height and width? I didn't see these params in WindowTraits class, but I guess there's exist a solution.
2. Is it possible to render window content during window resizing or moving? I know it's a kind of a problem on Windows platforms and I guess we can make some `resize_swapchain()` function with pure Vulkan (with some efforts). But does vsg has this ability or could it be written with code?

Anyway, I'll glad to see all replies. Thanks!

**Comment by robertosfield** [comment]
1. There isn't a mechanism directly in the VSG itself to set the minimum window size, so for this you'll need to get the native window handle and set this yourself.

2. The window resize normally just works out of the box, but I've just tried vsgviewer on my Windows 11 machine and it looks like the Windowing system is doing the resize by stretching the last frame rendered to the new window dimension then finally letting the VSG have a look in and giving the opportunity to properly resize.  Are you working under Windows 11 as well?

I can only put this down to yet another dumb arse thing that Microsoft does, perhaps there is a way to disable this so that window resize does interrupt the normal rendering, this is how other platforms work.  Hopefully fulltime Windows devs can chip in with insights on how to avoid this new Windows behavior.  It should be possible because vsgqtviewer works sensible.

**Reply by ComradeMashkov** [reply]
Thanks, Robert. I'm working under Windows 10 and it seems that this is the problem for all Windows versions. I know that other platforms don't have this irritating problem, so I'm still looking for a solution for Windows. I will be glad too if someone who is more professional than me give me an advice for this dumb thing :)
